python/pyspark/sql/functions.py

```diff
@@ -3300,6 +3300,88 @@ def map_zip_with(col1, col2, f):
     return _invoke_higher_order_function("MapZipWith", [col1, col2], [f])
 
 
+# ---------------------- Partition transform functions --------------------------------
+
+@since(3.1)
+def years(col):
+    """
+    Partition transform function: A transform for timestamps and dates
```

Comment:
Let's also clarify this expression should only with `partitionedBy` in DSv2 APIs.